File: cis_interface/examples/formatted_io5/src/formatted_io5_modelB.m
% Initialize input/output channels 
in_channel = CisInterface('CisPlyInput', 'inputB');
out_channel = CisInterface('CisPlyOutput', 'outputB');

flag = true;

% Loop until there is no longer input or the queues are closed
while flag

  % Receive input from input channel
  % If there is an error, the flag will be False.
  [flag, ply] = in_channel.recv();
  if (~flag)
    disp('Model B: No more input.');
    break;
  end;

  % Print received message
  fprintf('Model B: (%d verts, %d faces)\n', ply.nvert, ply.nface);
  fprintf('  Vertices:\n');
  for i = 1:ply.nvert
    fprintf('   %f, %f, %f\n', ...
	    ply['vertices'][i][0], ply['vertices'][i][1], ply['vertices'][i][2]);
  end;
  fprintf('  Faces:\n');
  for i = 1:ply.nface
    fprintf('   %d, %d, %d\n', ...
	    ply['faces'][i][0], ply['faces'][i][1], ply['faces'][i][2]);
  end;

  % Send output to output channel
  % If there is an error, the flag will be False
  flag = out_channel.send(ply);
  if (~flag)
    disp('Model B: Error sending output.');
    break;
  end;
  
end;

exit(0);

File: cis_interface/examples/formatted_io5/src/formatted_io5_modelB.py
import pprint
# Import classes for input/output channels
from yggdrasil.interface.YggInterface import (
    YggPlyInput, YggPlyOutput)

# Initialize input/output channels
in_channel = YggPlyInput('inputB')
out_channel = YggPlyOutput('outputB')

# Loop until there is no longer input or the queues are closed
while True:

    # Receive input from input channel
    # If there is an error, the flag will be False
    flag, ply = in_channel.recv()
    if not flag:
        print("Model B: No more input.")
        break

    # Print received message
    print('Model B: (%d verts, %d faces)' % (ply.nvert, ply.nface))
    pprint.pprint(ply)

    # Send output to output channel
    # If there is an error, the flag will be False
    flag = out_channel.send(ply)
    if not flag:
        raise RuntimeError("Model B: Error sending output.")
